http/handler.go

```diff
@@ -559,6 +560,13 @@ func handleRequestForwarding(core *vault.Core, handler http.Handler) http.Handle
 			return
 		}
 
+		// These requests should always be served locally
```

Comment:
This is an alternate approach compared to #7330 's way of not forwarding a specific authenticated request. With this approach, we don't have to expose a handler for this path prefix in handler.go like we're doing with host-info, but it means that we have to special case the path here (i.e. `handleRequestForwarding`) so that we don't end up forwarding the request. The check alone in `handleLogicalInternal` wouldn't suffice since it would go through this middleware first for any `v1/sys/` path with how the handler is set up for this route.

I'd like to get some group consensus on which the better approach this before merging.